File: TiAires/app/src/main/java/net/teamtruta/tiaires/GeoCacheAttributeListAdapter.java
package net.teamtruta.tiaires;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import java.util.ArrayList;
import java.util.List;

public class GeoCacheAttributeListAdapter extends BaseAdapter {

    private final List<GeoCacheAttributeEnum> attributes;
    private final LayoutInflater layoutInflater;
    private final Context context;

    public GeoCacheAttributeListAdapter(Context context, List<GeoCacheAttributeEnum> attributes) {
        this.attributes = attributes;
        this.layoutInflater = LayoutInflater.from(context);
        this.context = context;
    }


    @Override
    public int getCount() {
        return attributes.size();
    }

    @Override
    public Object getItem(int position) {
        return attributes.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        ViewHolder holder;

        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.attribute_element_layout, parent, false);
            holder = new ViewHolder();
            holder.attributeIcon = convertView.findViewById(R.id.attribute_icon);
            holder.attributeText = convertView.findViewById(R.id.attribute_text);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        int attributeIconID = GeoCacheAttributeIcon.Companion.getGeoCacheAttributeIcon(attributes.get(position));
        holder.attributeIcon.setImageDrawable(
                ContextCompat.getDrawable(context, attributeIconID));
                // DEPRECATED:
                //context.getDrawable(attributeIconID));
        holder.attributeText.setText(attributes.get(position).getAttributeString());

        return convertView;
    }

    static class ViewHolder {
        ImageView attributeIcon;
        TextView attributeText;
    }
}

File: TiAires/app/src/main/java/net/teamtruta/tiaires/adapters/GeoCacheAttributeListAdapter.kt
package net.teamtruta.tiaires.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import net.teamtruta.tiaires.extensions.GeoCacheAttributeIcon.Companion.getGeoCacheAttributeIcon
import net.teamtruta.tiaires.R
import net.teamtruta.tiaires.data.models.GeoCacheAttributeEnum

class GeoCacheAttributeListAdapter(private val context: Context,
                                   private val attributes: List<GeoCacheAttributeEnum>) : BaseAdapter() {
    private val layoutInflater: LayoutInflater = LayoutInflater.from(context)

    override fun getCount(): Int {
        return attributes.size
    }

    override fun getItem(position: Int): Any {
        return attributes[position]
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup): View? {
        var convertView = view
        val holder: ViewHolder
        if (convertView == null) {
            convertView = layoutInflater.inflate(R.layout.element_attribute, parent, false)
            holder = ViewHolder()
            holder.attributeIcon = convertView.findViewById(R.id.attribute_icon)
            holder.attributeText = convertView.findViewById(R.id.attribute_text)
            convertView.tag = holder
        } else {
            holder = convertView.tag as ViewHolder
        }
        val attributeIconID = getGeoCacheAttributeIcon(attributes[position])
        holder.attributeIcon!!.setImageDrawable(
                ContextCompat.getDrawable(context, attributeIconID))
        // DEPRECATED:
        //context.getDrawable(attributeIconID));
        holder.attributeText!!.text = attributes[position].attributeString
        return convertView
    }

    internal class ViewHolder {
        var attributeIcon: ImageView? = null
        var attributeText: TextView? = null
    }

}